src/web/rustdoc.rs

```diff
@@ -301,21 +291,60 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
     let latest_version = crate_details.latest_version().to_owned();
     let is_latest_version = latest_version == version;
 
-    let path = if !is_latest_version {
-        req_path[2] = &latest_version;
-        path_for_version(&req_path, &crate_details.target_name, &conn)
+    let path_in_latest = if !is_latest_version {
+        let mut latest_path = req_path.clone();
+        latest_path[2] = &latest_version;
+        path_for_version(&latest_path, &crate_details.target_name, &conn)
     } else {
         Default::default()
     };
 
+    // for each target the documentation is built for, check if the requested file also exists in
+    // that target and generate a list of targets + paths to render.
+    if let Some(Json::Array(doc_targets)) = &crate_details.doc_targets {
+        content.platforms = doc_targets
+            .iter()
+            .filter_map(|target| {
+                if let Json::String(target) = target {
+                    // rustdoc/:crate/:version[/:current_target]/:path*
+                    let mut platform_path = req_path.clone();
+                    // rustdoc/:crate/:version/:target/:path*
+                    if platform_path[3] == crate_details.target_name {
```

Comment:
Ugh, the naming in crate_details is so confusing.
```suggestion
                    //  `target_name` is the name of the crate, not the name of the platform.
                    if platform_path[3] == crate_details.target_name {
```